packages/cli/src/services/redis-client.service.ts

```diff
@@ -118,32 +155,29 @@ export class RedisClientService {
 	 * Reset the cumulative timeout if >30s between reconnection attempts.
 	 */
 	private retryStrategy() {
-		const RETRY_INTERVAL = 500; // ms
-		const RESET_LENGTH = 30_000; // ms
-		const MAX_TIMEOUT = this.globalConfig.queue.bull.redis.timeoutThreshold;
-
 		let lastAttemptTs = 0;
 		let cumulativeTimeout = 0;
 
 		return () => {
 			const nowTs = Date.now();
 
-			if (nowTs - lastAttemptTs > RESET_LENGTH) {
+			if (nowTs - lastAttemptTs > this.config.resetLength) {
 				cumulativeTimeout = 0;
 				lastAttemptTs = nowTs;
 			} else {
 				cumulativeTimeout += nowTs - lastAttemptTs;
 				lastAttemptTs = nowTs;
-				if (cumulativeTimeout > MAX_TIMEOUT) {
-					this.logger.error(`[Redis] Unable to connect after max timeout of ${MAX_TIMEOUT} ms`);
-					this.logger.error('Exiting process...');
+				if (cumulativeTimeout > this.config.maxTimeout) {
+					const maxTimeout = Math.round(this.config.maxTimeout / 1000) + 's';
+					this.logger.error(`Unable to connect to Redis after trying to connect for ${maxTimeout}`);
+					this.logger.error('Exiting process due to Redis connection error');
 					process.exit(1);
```

Comment:
Ideally we should allow graceful exit of other parts of the app as well. Otherwise we might leave some resources in improper state, like DB connections etc. Just mentioning this, no need to act on it here